src/browser/ui/dom/DOMChildrenOperations.js

```diff
@@ -31,10 +31,21 @@ function insertChildAt(parentNode, childNode, index) {
   // rely exclusively on `insertBefore(node, null)` instead of also using
   // `appendChild(node)`. However, using `undefined` is not allowed by all
   // browsers so we must replace it with `null`.
-  parentNode.insertBefore(
-    childNode,
-    parentNode.childNodes[index] || null
-  );
+
+  // fix render order error in safari
+  if (!(document.all && !document.addEventListener)) {
```

Comment:
As far as I can tell, you simply can't index outside the size of the list, so something like `index < parentNodes.childNodes.length ? parentNodes.childNodes[index] : null` should be enough.